src/stream_processor/flb_sp.c

```diff
@@ -526,6 +523,319 @@ struct flb_sp *flb_sp_create(struct flb_config *config)
+        } else if (val.type == MSGPACK_OBJECT_FLOAT32 ||
+                   val.type == MSGPACK_OBJECT_FLOAT) {
+           flb_info("2");
+            result = flb_malloc(sizeof(struct flb_exp_val));
+            result->type = FLB_EXP_FLOAT;
+            result->val.f64 = val.via.f64;
+            return result;
+        } else if (val.type == MSGPACK_OBJECT_STR) {
+            flb_info("3");
+            result = flb_malloc(sizeof(struct flb_exp_val));
+
+            result->type = FLB_EXP_STRING;
+            result->val.string = flb_sds_create_len((char *) val.via.str.ptr, val.via.str.size);
+            return result;
+        }
+    }
+
+    return NULL;
+}
+
+void numerical_comp(struct flb_exp_val *left, struct flb_exp_val *right, struct flb_exp_val *result, int op)
+{
+    result->type = FLB_EXP_BOOL;
+
+    if (left == NULL || right == NULL) {
+        result->val.boolean = false;
+        return;
+    }
+
+    flb_info("2 - operation: %d", op);
+    switch (op) {
+        case FLB_EXP_EQ:
+            if (left->type == right->type) // TODO: we should be able to compare i64 and f64 numbers
+            {   switch(left->type){
+                    case FLB_EXP_BOOL:
+                        result->val.boolean = (left->val.boolean == right->val.boolean);
+                        break;
+                    case FLB_EXP_INT:
+                        result->val.boolean = (left->val.i64 == right->val.i64);
+                        break;
+                    case FLB_EXP_FLOAT:
+                        result->val.boolean = (left->val.f64 == right->val.f64);
+                        break;
+                    case FLB_EXP_STRING:
+                        flb_info("string operation: %s %s", left->val.string, right->val.string);
+                        if (flb_sds_len(left->val.string) != flb_sds_len(right->val.string))
+                            result->val.boolean = false;
+                        else if (strncmp(left->val.string, right->val.string, flb_sds_len(left->val.string)) != 0)
+                            result->val.boolean = false;
+                        else
+                            result->val.boolean = true;
+                        break;
+                    default:
+                        result->val.boolean = false;
+                        break;
+                }
+            } else
+                result->val.boolean = false;
+            break;
+        case FLB_EXP_LT:
+            if (left->type == right->type)
+            {   switch(left->type){
+                    case FLB_EXP_INT:
+                        result->val.boolean = (left->val.i64 < right->val.i64);
+                        break;
+                    case FLB_EXP_FLOAT:
+                        result->val.boolean = (left->val.f64 < right->val.f64);
+                        break;
+                    case FLB_EXP_STRING:
+                        if (strncmp(left->val.string, right->val.string, flb_sds_len(left->val.string)) < 0)
```

Comment:
please add braces: https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#braces-usage-on-conditionals-loops-and-functions